### PR TITLE
 Remove command-line interface from emscripten.py

### PR DESCRIPTION
**Pull request by dschuff**
There are no more uses of emscripten.py other than being called directly from `Shared.Building.emscripten`. So stop building and then parsing a command line.

### CODE REVIEW

**Comment by kripken** [comment]
cc @waywardmonkeys - are you aware of an out-of-tree direct use of emscripten.py?

**Comment by waywardmonkeys** [comment]
We used to ... long ago. I removed that from our builds in favor of directly invoking `clang` and then using `emcc` as the link step long ago. (2014?)

**Comment by kripken** [comment]
Great, thanks @waywardmonkeys!

File: emscripten.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def emscript(infile, outfile, libraries, compiler_engine, temp_files,
 
   Args:
     infile: The path to the input LLVM assembly file.
-    outfile: The file where the output is written.
+    outfile: An open file object where the output is written.
```
Comment (jgravelle-google) [inline emscripten.py]: Yay type annotations <_<

File: tools/shared.py [hunk 2]
```diff
@@ -2267,16 +2267,18 @@ def emar(action, output_filename, filenames, stdout=None, stderr=None, env=None)
...
     if path_from_root() not in sys.path:
       sys.path += [path_from_root()]
     import emscripten
     # Run Emscripten
-    cmdline = [filename + ('.o.ll' if append_ext else ''), '-o', filename + '.o.js'] + extra_args
+    infile = filename
+    outfile = filename + '.o.js'
     if jsrun.TRACK_PROCESS_SPAWNS:
-      logging.info('Executing emscripten.py compiler with cmdline "' + ' '.join(cmdline) + '"')
+      logging.info('Executing emscripten.py compiler with cmdline "' + ' '.join([infile, outfile]) + '"')
```
Comment (jgravelle-google) [inline tools/shared.py]: Does this log even make sense now? It's not tracking a process spawn, and it's being called with args, not a cmdline.

**Comment by dschuff** [comment]
I'm here to delete code and chew bubblegum; and I'm all out of bubblegum.

**Pull request merged**